File: Determine_if_Two_Strings_Are_Close.java
class Solution {
    public boolean closeStrings(String word1, String word2) {
        if (word1.length() != word2.length())
        return false;

        Map<Character, Integer> w1 = new HashMap<>();
        Map<Character, Integer> w2 = new HashMap<>();

        for (final char c : word1.toCharArray())
        w1.merge(c, 1, Integer::sum);

        for (final char c : word2.toCharArray())
        w2.merge(c, 1, Integer::sum);

        if (!w1.keySet().equals(w2.keySet()))
        return false;

        List<Integer> freqs1 = new ArrayList<>(w1.values());
        List<Integer> freqs2 = new ArrayList<>(w2.values());

        Collections.sort(freqs1);
        Collections.sort(freqs2);
        return freqs1.equals(freqs2);        
    }
}

File: Determine_if_Two_Strings_Are_Close.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:
        if len(word1) != len(word2):
            return False
        
        w1 = {}
        for c in word1:
            w1[c] = 1 + w1.get(c, 0)

        w2 = {}
        for c in word2:
            w2[c] = 1 + w2.get(c, 0)
        
        if w1.keys() != w2.keys():
            return False
        
        return sorted(w1.values()) == sorted(w2.values())
